### PR TITLE
load more messages into thread in serial

### PR DESCRIPTION
**Pull request by mmaxim**
@chrisnojima what do you think of this? Seems like this will line up all requests to modify threads so that we don't get in a situation where two sagas are trying to update at the same time. Really this will just guard against all scenarios where we issue two `loadMoreMessages` calls for the same convo (I don't know why that happens).

### CODE REVIEW

File: shared/actions/chat/thread-content.js [hunk 1]
```diff
@@ -744,24 +744,21 @@ function* _updateThread({
...
-
-function* _prependMessagesToConversation({
-  payload: {conversationIDKey, messages},
-}: ChatGen.AppendMessagesPayload) {
-  const currentMessages = yield Saga.select(Constants.getConversationMessages, conversationIDKey)
-  const nextMessages = I.OrderedSet(messages.map(m => m.key)).concat(currentMessages)
-  yield Saga.put(
-    EntityCreators.replaceEntity(['conversationMessages'], I.Map({[conversationIDKey]: nextMessages}))
-  )
+function* _appendOrPrependMessagesToConversation(arg: ChatGen.AppendMessagesPayload) {
```
Comment (chrisnojima) [inline shared/actions/chat/thread-content.js]: not your bug but really should be
 ```
ChatGen.AppendMessagesPayload | ChatGen. PrependMessagesPayload
```

File: shared/actions/chat/thread-content.js [hunk 2]
```diff
@@ -744,24 +744,21 @@ function* _updateThread({
...
+  const {conversationIDKey, messages} = arg.payload
+  if (arg.type === ChatGen.appendMessages) {
+    const currentMessages = yield Saga.select(Constants.getConversationMessages, conversationIDKey)
```
Comment (chrisnojima) [inline shared/actions/chat/thread-content.js]: really currentMessages and the Saga.put are the same so really your if just has to be the const nextMessages lines

**Comment by mmaxim** [comment]
This doesn't seem to fix the problem, which I can repro fairly reliably now. I think this might just be a bug when rendering the thread, not fetching it into this ordered set? Two things clear the bad messages:

1.) Scrolling up to the next page
2.) Backing out and into the thread (on mobile).

It seems like basically anything that makes it redraw clears the bad messages out of there. Does that ring any bells on something that may have changed recently?

**Comment by mmaxim** [comment]
Nevermind, the store is wrong somehow.

**Comment by mmaxim** [comment]
This is not the correct fix to the problem, and in fact makes the app much worse, closing.

**Pull request closed without merging**